File: lib/app/database/email_dao.dart
import 'package:email_alias/app/database/email.dart';
import 'package:floor/floor.dart';

@dao
abstract class EmailDao {
  @Query('SELECT * FROM Email')
  Future<List<Email>> getAll();

  @Query('SELECT * FROM Email')
  Stream<List<Email>> getAllStream();

  @insert
  Future<void> insertEmail(final Email email);

  @insert
  Future<void> insertEmails(final List<Email> emails);

  @update
  Future<void> updateEmail(final Email email);

  @delete
  Future<void> deleteEmail(final Email email);
  
  @Query('DELETE FROM Email')
  Future<void> deleteAll();
}

File: android/email_alias_watch/src/main/java/com/opdehipt/email_alias/database/EmailDao.kt
package com.opdehipt.email_alias.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface EmailDao {
    @Query("SELECT * FROM email")
    suspend fun getAll(): List<Email>

    @Query("SELECT * FROM email")
    fun getAllFlow(): Flow<List<Email>>

    @Query("SELECT * FROM email WHERE id = :id")
    suspend fun get(id: Int): Email

    @Insert
    suspend fun insertEmail(email: Email)

    @Insert
    suspend fun insertEmails(emails: List<Email>)

    @Update
    suspend fun updateEmail(email: Email)

    @Delete
    suspend fun deleteEmail(email: Email)

    @Query("DELETE FROM Email")
    suspend fun deleteAll()
}